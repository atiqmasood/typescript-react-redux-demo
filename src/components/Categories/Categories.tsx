import * as React from "react";

import { Button, Dropdown, Grid } from "semantic-ui-react";
/**
 * Components interfaces/types
 */
export interface IProductIds {
  key: number;
  value: number;
  text: number;
}

export interface IProps {
  children?: React.ReactNode;
  getProductIds?: any;
  productIds?: IProductIds[];
  getProducts?: any;
}

class Categories extends React.Component<IProps> {
  public productIdNode: any;
  /**
   * Components constructors with basic initializations.
   * @param props
   */
  constructor(props: IProps) {
    super(props);

    this.state = {};
    this.productIdNode = React.createRef();
  }
  /**
   * get products when on click get products button
   */
  public getProducts = (event: any) => {
    const { getProducts } = this.props;
    const productIds = this.productIdNode.current.state.value || [];
    getProducts(productIds);
  };
  public render() {
    const { productIds } = this.props;

    return (
      <React.Fragment>
        <Grid columns="equal">
          <Grid.Column>
            <Dropdown
              ref={this.productIdNode}
              placeholder="Choose product ids"
              fluid={true}
              multiple={true}
              search={true}
              selection={true}
              options={productIds}
            />
          </Grid.Column>
          <Grid.Column>
            <Button basic={true} color="violet" onClick={this.getProducts}>
              Get Products
            </Button>
          </Grid.Column>
        </Grid>
      </React.Fragment>
    );
  }
}

export default Categories;
