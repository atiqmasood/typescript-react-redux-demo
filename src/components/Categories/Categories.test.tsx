import * as React from "react";
import Categories from "./Categories";

import * as renderer from "react-test-renderer";
/**
 * Test suites for categories/ productids
 */
describe("Categories component renders correctly", () => {
  it("Renders Correctly", () => {
    const productIds = [
      {
        key: 3383096,
        value: 3383096,
        text: 3383096
      }
    ];
    const rendered = renderer.create(<Categories productIds={productIds} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
