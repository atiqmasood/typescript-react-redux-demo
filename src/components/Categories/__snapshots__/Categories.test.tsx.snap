// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component renders correctly Renders Correctly 1`] = `
<div
  className="ui equal width grid"
>
  <div
    className="column"
  >
    <div
      aria-busy={undefined}
      aria-disabled={undefined}
      aria-expanded={false}
      className="ui fluid multiple search selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={undefined}
    >
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex={0}
        type="text"
        value=""
      />
      <span
        className="sizer"
      />
      <div
        aria-live="polite"
        className="default text"
        role="alert"
      >
        Choose product ids
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        aria-multiselectable={true}
        className="menu transition"
        role="listbox"
      >
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            3383096
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="column"
  >
    <button
      className="ui violet basic button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Get Products
    </button>
  </div>
</div>
`;
